tests/on-device/.run.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -i bash -p bats
+
+# Copyright (c) 2019-2021, see AUTHORS. Licensed under MIT License, see LICENSE.
+
+set -ueo pipefail
+
+CONFIRMATION_FILE="$HOME/.cache/nix-on-droid-self-test/confirmation-granted"
+SCRIPT_DIR="$(dirname "$(readlink -f "$0")")"
+if [[ ! -e "$CONFIRMATION_FILE" ]]; then
+    echo 'These semi-automated tests are destructive!'
+    echo 'They are meant to be executed by maintainers on a clean install.'
+    echo 'Proceeding will wreck your installation.'
+    echo 'Do you still wish to proceed?'
+    echo -n '(type "I do" to proceed, anything else to abort) > '
+    read -r CONFIRMATION
+    if [[ "$CONFIRMATION" != 'I do' ]]; then echo 'Cool, aborting.'; exit 7; fi
+    touch "$CONFIRMATION_FILE"
+fi
+
+if [[ ! -d ~/.config.bak ]]; then
+    mv ~/.config ~/.config.bak
+    cp -r ~/.config.bak ~/.config
+fi
+
+pushd "$SCRIPT_DIR"  # for lib.sh import
```

Comment:
You can replace the pushd stuff with this nice feature to load shared code: https://bats-core.readthedocs.io/en/stable/writing-tests.html#load-share-common-code